lsp-metals.el

```diff
@@ -200,9 +223,39 @@ displayed either as additional decorations."
    ("metals.show-inferred-type" lsp-metals-show-inferred-type t)
    ("metals.remote-language-server" lsp-metals-remote-language-server)))
 
+(lsp-dependency
+ 'coursier
+ '(:system "coursier")
+ `(:download :url lsp-metals-coursier-download-url
+             :store-path lsp-metals-coursier-store-path
+             :set-executable? t))
+
+(lsp-dependency
+ 'metals
+ `(:system ,lsp-metals-server-command)
+ `(:system ,(f-join lsp-metals-server-install-dir "metals")))
+
 (defun lsp-metals--server-command ()
   "Generate the Scala language server startup command."
-  `(,lsp-metals-server-command ,@lsp-metals-server-args))
+  `(,(lsp-package-path 'metals) ,@lsp-metals-server-args))
+
+(defun lsp-metals--download-server (_client callback error-callback _update?)
+  "Install metals server via coursier.
+Will invoke CALLBACK on success, ERROR-CALLBACK on error."
+  (lsp-package-ensure
+   'coursier
+   (lambda ()
+     (call-process
+      (lsp-package-path 'coursier)
+      nil
+      (get-buffer-create "*Coursier log*")
+      t
+      "install"
```

Comment:
So one thing maybe to think about is that by using install rather than bootstrap, the user can only install a stable version. Am I understanding all this correct that if someone wants to use a snapshot, they need to take care of it themselves? It _may_ be pretty simple to instead of using install, just use bootstrap, which would allow the user to then specify the version if they want a snapshot, and if not, then just give them the latest. That's what I ended up doing in nvim-metals https://github.com/scalameta/nvim-metals/blob/381fd8c1a594a6b4ae7dd78d9b4522fc969beb15/lua/metals/setup.lua#L71-L113 and it works pretty well. But take it or leave it.

Reply:
Added `lsp-metals-install-version` variable.

Reply:
Updated MR to use `bootstrap`.